### DISCUSSION TITLE
What should a self hosted anti spam solution look like?

### DISCUSSION THREAD

**Discussion by TLWebdesign** [joomla/joomla-cms#42044] [General]
Hi,

I start this discussion to see what you guys think a self hosted form antis-spam solution should ook like, what tactics it should use and what it should not? 

Any other requirements you can think of. I'd like to come up with a list of functionality we would want. Maybe from there i'll be able to get something of the ground.

**Comment by TLWebdesign** [comment]
My idea's for a simple anti spam plugin:

- Configurable Honeypot field 
- Configurable minimum time for filling out the form

Features: 
- configurable (but fallback to randomized) css class and honeypot field name. 
- configurable minimum time
- exclude components and/or views

Any feedback appreciated.

**Reply by brianteeman** [reply]
what do you mean by time limit

**Reply by TLWebdesign** [reply]
The time it took for the form to be filled in. if it is below the time limit it would detect that as bot activity.

**Reply by brianteeman** [reply]
so a minimum time below which it fails rather than a maximum

**Reply by TLWebdesign** [reply]
yes thats correct.

**Comment by micker** [comment]
maybe you can this : https://github.com/joomla/joomla-cms/discussions/39662
